versioned/storage/common-tests/src/main/java/org/projectnessie/versioned/storage/commontests/AbstractIndexesLogicTests.java

```diff
@@ -338,4 +342,79 @@ public void perCommitOps() {
 
     soft.assertThat(indexesLogic.commitOperations(commit)).containsExactly(add1, add2, remove1);
   }
+
+  @SuppressWarnings("unused")
+  static StoreConfig.Adjustable persistWithSmallIndexSize(StoreConfig.Adjustable config) {
+    return config.withMaxIncrementalIndexSize(200);
+  }
+
+  @Test
+  void completeIndexesInCommitChainWithStripes(
+      @NessiePersist(configMethod = "persistWithSmallIndexSize") Persist persist) throws Exception {
+
+    ObjId currentId = fiveIncompleteCommitsWithThreeOpsEach(persist);
+
+    IndexesLogic indexesLogic = indexesLogic(persist);
+    indexesLogic.completeIndexesInCommitChain(currentId, () -> {});
+
+    for (int i = 5; i > 0; i--) {
+      CommitObj current = persist.fetchTypedObj(currentId, ObjType.COMMIT, CommitObj.class);
+      if (i > 1) {
+        // All commits except commit 1 should have stripes
+        soft.assertThat(current.referenceIndexStripes()).isNotEmpty();
+      }
+      int actualElementCount =
+          indexesLogic.incrementalIndexFromCommit(current).elementCount()
+              + (current.hasReferenceIndex()
+                  ? requireNonNull(indexesLogic.buildReferenceIndexOnly(current)).elementCount()
+                  : 0);
+      int expectedElementCount = i * 3;
+      soft.assertThat(actualElementCount).isEqualTo(expectedElementCount);
+      currentId = current.directParent();
+    }
+  }
+
+  private ObjId fiveIncompleteCommitsWithThreeOpsEach(Persist persist) throws ObjTooLargeException {
+
+    String[][] keys =
+        new String[][] {
+          {"abcd", "bcde", "cdef"},
+          {"efgh", "fghi", "ghij"},
+          {"hijk", "ijkl", "jklm"},
+          {"klmn", "lmno", "mnop"},
+          {"nopq", "opqr", "pqrs"},
+        };
+
+    CommitObj current = null;
+    for (int i = 0; i < 5; i++) {
+
+      ObjId parentId = current == null ? EMPTY_OBJ_ID : current.id();
+
+      CommitObj.Builder c =
+          commitBuilder()
+              .id(randomObjId())
+              .addTail(parentId)
+              .created(System.nanoTime())
```

Comment:
Nit: its microseconds since epoch, better fix it here, before some copies it elsewhere.

Or just add some constant timestamps like `TimeUnit.SECONDS.toMicros(i + 1)` or so.